evoGUI.lua

```diff
@@ -23,12 +23,49 @@ function evogui.mod_init()
     end
 end
 
-
 function evogui.mod_update(data)
-    if data.mod_changes["{{MOD_NAME}}"] then
-        -- TODO: If a more major migration ever needs doing, do that here.
-        -- Otherwise, just falling back to mod_init should work fine.
-        evogui.mod_init()
+    if data.mod_changes then
+        if data.mod_changes["{{MOD_NAME}}"] then
+            -- TODO: If a more major migration ever needs doing, do that here.
+            -- Otherwise, just falling back to mod_init should work fine.
+            evogui.mod_init()
+        end
+
+        evogui.validate_sensors(data.mod_changes)
+    end
+end
+
+-- Iterate through all value_sensors, if any are associated with a mod_name that
+-- has been removed, remove the sensor from the list of value_sensors.
+function evogui.validate_sensors(mod_changes)
+    local valid_sensors = {}
+    for sensor in evogui.value_sensors do
+        if sensor.mod_name and mod_changes[sensor.mod_name] then
+            -- mod upgrade, keep sensor
+            if mod_changes[sensor.mod_name].new_version ~= nil then
+                valid_sensors[#valid_sensors + 1] = sensor
```

Comment:
Agreed on the table insert, but I actually think I might be able to rewrite this to be a bit briefer, and not create a new table copy `valid_sensors` at all. I did a bit of research, and In Lua, it is not safe to iterate a table forward and remove elements (which is why I made the copy), but apparently it is safe to iterate a table *backwards* and remove elements. That would shorten this code by several lines.